test/ibmq/test_ibmq_qasm_simulator.py

```diff
@@ -133,14 +136,21 @@ def _new_submit(qobj, *args, **kwargs):
         qobj = assemble(circ, backend=backend, header={'test': 'qobj'})
         backend.run(qobj)
 
+        backend._configuration._data['simulation_method'] = sim_method
+        backend._submit_job = submit_fn
```

Comment:
You should put the restoration in a `try...finally` block otherwise this code won't get executed if the test hits an exception.